content/automate/ManageIQ/Transformation/Common.class/__methods__/acquiretransformationhost.rb

```diff
@@ -5,27 +5,14 @@ module Common
         class AcquireTransformationHost
           def initialize(handle = $evm)
             @handle = handle
+            @task = ManageIQ::Automate::Transformation::Common::Utils.task(@handle)
           end
 
           def main
-            factory_config = @handle.get_state_var(:factory_config)
-            raise "No factory config found. Aborting." if factory_config.nil?
-
-            task = @handle.root['service_template_transformation_plan_task']
-            @handle.log(:info, "Task: #{task.inspect}")
-
-            transformation_host_type, transformation_host, transformation_method = ManageIQ::Automate::Transformation::TransformationHosts::Common::Utils.get_transformation_host(task, factory_config)
-            if transformation_host.nil?
-              @handle.log(:info, "No transformation host available. Retrying.")
+            if @task.get_option(:transformation_host_id).nil?
               @handle.root['ae_result'] = 'retry'
               @handle.root['ae_retry_server_affinity'] = true
-              @handle.root['ae_retry_interval'] = $evm.object['check_convert_interval'] || '1.minutes'
-            else
-              @handle.log(:info, "Transformation Host: #{transformation_host.name}.")
-              task.set_option(:transformation_host_id, transformation_host.id)
-              task.set_option(:transformation_host_name, transformation_host.name)
-              task.set_option(:transformation_host_type, transformation_host_type)
-              task.set_option(:transformation_method, transformation_method)
+              @handle.root['ae_retry_interval'] = 15.seconds
```

Comment:
does it make sense to have an overridable value `e.g. $evm.object['...'] || 15.seconds`
(honest question)

Reply:
Using `$evm.object['...']` means adding an attribute to the class. And the class is really generic. I'd defer that to another key/value store, such as `$evm.get_state_var('...')` or even data from the throttler request/task itself, as it sets the tempo for host allocation.